src/inference/src/threading/ie_istreams_executor.cpp

```diff
@@ -46,6 +46,57 @@ int IStreamsExecutor::Config::GetDefaultNumStreams() {
         return 1;
 }
 
+int IStreamsExecutor::Config::GetHybridNumStreams(const Config& config, const int stream_mode) {
+    const int num_cores = parallel_get_max_threads();
+    const int num_phy_cores = getNumberOfCPUCores();
+    const int num_big_cores = getNumberOfCPUCores(true);
+    const int num_small_cores = num_phy_cores - num_big_cores;
+
+    if (stream_mode == DEFAULT) {
+        // bare minimum of streams (that evenly divides available number of core)
+        if (0 == num_big_cores % 4) {
+            config._big_core_streams = std::max(4, num_big_cores / 4);
+        } else if (0 == num_big_cores % 5) {
+            config._big_core_streams = std::max(5, num_big_cores / 5);
+        } else if (0 == num_big_cores % 3) {
+            config._big_core_streams = std::max(3, num_big_cores / 3);
+        } else {  // if user disables some cores say in BIOS, so we got weird #cores which is not easy to divide
+            config._big_core_streams = 1;
+        }
+
+        config._threads_per_stream_big = num_big_cores / config._big_core_streams;
+        config._threads_per_stream_small = config._threads_per_stream_big * 2;
+        if (num_small_cores == 0) {
+            config._big_core_streams = num_big_cores / config._threads_per_stream_big;
+            config._threads_per_stream_small = 0;
+        } else if (num_small_cores < config._threads_per_stream_small) {
+            config._small_core_streams = 1;
+            config._threads_per_stream_small = num_small_cores;
+            config._threads_per_stream_big = config._threads_per_stream_small / 2;
+            config._big_core_streams = num_big_cores / config._threads_per_stream_big;
+        } else {
+            config._small_core_streams = num_small_cores / config._threads_per_stream_small;
+        }
+    } else if (stream_mode == AGGRESSIVE) {
+        config._big_core_streams = num_big_cores;
+        config._small_core_streams = num_small_cores / 2;
+        config._threads_per_stream_big = num_big_cores / config._big_core_streams;
+        config._threads_per_stream_small = num_small_cores == 0 ? 0 : num_small_cores / config._small_core_streams;
+    } else if (stream_mode == LESSAGGRESSIVE) {
+        config._big_core_streams = num_big_cores / 2;
```

Comment:
In LESSAGGRESSIVE, please use `config._big_core_streams = num_big_cores` and `config._small_core_streams = num_small_cores /2` by default